azurelinuxagent/common/protocol/wire.py

```
@@ -1024,18 +1031,45 @@ def send_request_using_appropriate_channel(self, direct_func, host_func):
+                  "Error: {2}".format(old_container_id, old_role_config_name, ustr(e))
             logger.info(msg)
 
             self.update_goal_state(forced=True)
+
+            new_container_id = self.host_plugin.container_id
+            new_role_config_name = self.host_plugin.role_config_name
             msg = "Host plugin reconfigured with new parameters. " \
-                  "ContainerId: {0}, role config file: {1}.".format(self.host_plugin.container_id,
-                                                                    self.host_plugin.role_config_name)
+                  "ContainerId: {0}, role config file: {1}.".format(new_container_id, new_role_config_name)
             logger.info(msg)
 
-            ret = host_func()
+            try:
+                ret = host_func()
+                if ret:
+                    msg = "Request succeeded using the host plugin channel after goal state refresh." \
+                          "ContainerId changed from {0} to {1}, " \
+                          "role config file changed from {2} to {3}.".format(old_container_id, new_container_id,
+                                                                             old_role_config_name, new_role_config_name)
+                    add_event(name=AGENT_NAME,
+                              version=CURRENT_VERSION,
+                              op=WALAEventOperation.HostPlugin,
+                              is_success=True,
+                              message=msg,
+                              log_event=False)
+
+            except Exception as e:
+                add_event(name=AGENT_NAME,
+                          version=CURRENT_VERSION,
+                          op=WALAEventOperation.HostPlugin,
+                          is_success=False,
+                          message=ustr(e),
+                          log_event=False)
+                raise e
+
```

Why is this needed?

```
         except Exception:
            raise
```

---

I needed a try/catch to emit a telemetry event, but originally we weren't swallowing the exceptions from the retry call.

---

But it is not needed. We aren't doing anything in the exception handler, just re-raising the exception.